src/main/java/com/google/api/codegen/discogapic/transformer/java/JavaDiscoGapicSchemaToViewTransformer.java

```diff
@@ -0,0 +1,195 @@
+    return new ModelTypeTable(
+        new JavaTypeTable(implicitPackageName),
+        new JavaModelTypeNameConverter(implicitPackageName));
+  }
+
+  private StaticLangApiSchemaFileView generateSchemaFile(DiscoGapicInterfaceContext context,
+      String schemaName, Schema schema, SymbolTable docSymbolTable) {
+    StaticLangApiSchemaFileView.Builder apiFile = StaticLangApiSchemaFileView.newBuilder();
+    // Escape any schema's field names that are Java keywords.
+    SymbolTable schemaSymbolTable = SymbolTable.fromSeed(JavaNameFormatter.RESERVED_IDENTIFIER_SET);
+
+    apiFile.templateFileName(SCHEMA_TEMPLATE_FILENAME);
+
+    apiFile.schema(generateSchemaClass(context, schemaName, schema, schemaSymbolTable));
+
+    String outputPath = pathMapper.getOutputPath(null, context.getProductConfig());
+    apiFile.outputPath(outputPath + File.separator + schemaName + ".java");
+
+    // must be done as the last step to catch all imports
+    apiFile.fileHeader(fileHeaderTransformer.generateFileHeader(context));
+
+    return apiFile.build();
+  }
+
+  private StaticLangApiSchemaView generateSchemaClass(DiscoGapicInterfaceContext context,
+      String schemaName, Schema schema, SymbolTable schemaSymbolTable) {
+    addApiImports(context);
+
+    StaticLangApiSchemaView.Builder schemaView = StaticLangApiSchemaView.newBuilder();
+
+    schemaView.typeName(schemaName);
+    schemaView.type(schema.type());
+    // TODO(andrealin): apply Java naming format.
+    // TODO(andrealin): use symbol table to make sure Schema names aren't Java keywords.
+    schemaView.className(schemaName);
+    schemaView.defaultValue(schema.defaultValue());
+
+    // Map each property name to the Java typeName of the property.
+    List<SimplePropertyView> properties = new LinkedList<>();
+    for (Map.Entry<String, Schema> propertyEntry : schema.properties().entrySet()) {
+      String propertyName = schemaSymbolTable.getNewSymbol(propertyEntry.getKey());
+      Schema property = propertyEntry.getValue();
+      SimplePropertyView.Builder simpleProperty = SimplePropertyView.newBuilder()
+          .name(propertyName).repeated(property.repeated());
+      simpleProperty.typeName(typeToJavaType(propertyName, property));
+      properties.add(simpleProperty.build());
+    }
+    schemaView.properties(properties);
+
+    return schemaView.build();
+  }
+
+  private void addApiImports(DiscoGapicInterfaceContext context) {
+    ModelTypeTable typeTable = context.getModelTypeTable();
+    typeTable.saveNicknameFor("com.google.api.core.BetaApi");
+    typeTable.saveNicknameFor("java.io.Serializable");
+    typeTable.saveNicknameFor("java.util.List");
+    typeTable.saveNicknameFor("javax.annotation.Generated");
+  }
+
+  // Return the corresponding Java identifier for a given Discovery doc typeName and format.
+  // https://developers.google.com/discovery/v1/type-format.
+  private String typeToJavaType(String name, Schema schema) {
```

Comment:
We should create something analogous to `JavaModelTypeNameConverter` for discovery types - it would be named something like `JavaDiscoTypeNameConverter`. Also, don't use `ModelTypeTable` (referenced above) because this is for processing the type model from protobuf. You will need a `DiscoTypeTable` (or `SchemaTypeTable`?)